Wrong step for Mat(std::vector) in 5.x

### System Information

Latest Debian, OpenCV 5.x at 2ed6d0f59012751cc99dd0dd29e614b17a59dcc1 (nov 19th)

### Detailed description

It seems the step is not properly computed for a matrix built using an `std::vector`

### Steps to reproduce

```cpp
TEST(Imgproc_Resize, vec)
{
  std::vector<float> vec(4);
  Mat A(vec), B;
  std::cout<<A.size()<<std::endl;
  resize(A,B,Size(2,2),0,0,INTER_LINEAR);
}
```
triggers:
```
[ RUN      ] Imgproc_Resize.vec
[4 x 1]
unknown file: Failure
C++ exception with description "OpenCV(5.0.0-pre) /usr/local/google/home/vrabaud/opencv5/src/opencv/modules/core/src/matrix.cpp:904: error: (-2:Unspecified error) in function 'cv::Mat::Mat(Size, int, void *, size_t)'
>  (expected: '_step >= minstep'), where
>     '_step' is 4
> must be greater than or equal to
>     'minstep' is 16
" thrown in the test body.
[  FAILED  ] Imgproc_Resize.vec (0 ms)

```

When `src.step` is cast to a `size_t` at https://github.com/opencv/opencv/blob/d0820dac382943cb35b7ca4a0d51458164010803/modules/imgproc/src/resize.cpp#L4244, the value 4 (sizeof(float)) is chosen instead of 16 (4*sizeof(float)). This is because of https://github.com/opencv/opencv/blob/d0820dac382943cb35b7ca4a0d51458164010803/modules/core/include/opencv2/core/mat.inl.hpp#L547

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [X] There is reproducer code and related data files (videos, images, onnx, etc)

--- a/modules/core/src/matrix.cpp
+++ b/modules/core/src/matrix.cpp
@@ -895,7 +895,7 @@ Mat::Mat(Size _sz, int _type, void* _data, size_t _step)
 
     size_t esz = CV_ELEM_SIZE(_type), esz1 = CV_ELEM_SIZE1(_type);
     size_t minstep = cols*esz;
-    if( _step == AUTO_STEP )
+    if( _step == AUTO_STEP || (_step < minstep && rows == 1))
     {
         _step = minstep;
     }

--- a/modules/imgproc/test/test_imgwarp.cpp
+++ b/modules/imgproc/test/test_imgwarp.cpp
@@ -1204,7 +1204,6 @@ TEST(Imgproc_Resize, issue_26497)
 {
     std::vector<float> vec = {0.f, 1.f, 2.f, 3.f};
     Mat A(vec), B;
-    std::cout << A.size() << std::endl;
     resize(A, B, Size(2,2), 0, 0, INTER_LINEAR);
     double minv = 0, maxv = 0;
     cvtest::minMaxIdx(B, &minv, &maxv, nullptr, nullptr, noArray());

--- a/modules/imgproc/test/test_imgwarp.cpp
+++ b/modules/imgproc/test/test_imgwarp.cpp
@@ -1200,5 +1200,18 @@ TEST(Imgproc_Remap, issue_23562)
     }
 }
 
+TEST(Imgproc_Resize, issue_26497)
+{
+    std::vector<float> vec = {0.f, 1.f, 2.f, 3.f};
+    Mat A(vec), B;
+    std::cout << A.size() << std::endl;
+    resize(A, B, Size(2,2), 0, 0, INTER_LINEAR);
+    double minv = 0, maxv = 0;
+    cvtest::minMaxIdx(B, &minv, &maxv, nullptr, nullptr, noArray());
+    EXPECT_EQ(B.size(), Size(2, 2));
+    EXPECT_LE(0., minv);
+    EXPECT_LE(maxv, 3.);
+}
+
 }} // namespace
 /* End of file. */

